### PR TITLE
[plugins/digitalocean][feat] Digital Ocean collector, part 1/2 - tests

### PR DESCRIPTION
**Pull request by meln1k**
# Description

Added some test for the DigitalOcean collector code.

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`
- [ ] Document new or updated functionality (someengineering/resoto.com#XXXX)

# Issues Fixed

- Fixes #XXXX

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/digitalocean/resoto_plugin_digitalocean/collector.py [hunk 1]
```diff
@@ -393,6 +394,14 @@ def collect_regions(self) -> None:
     @metrics_collect_volumes.time()
     def collect_volumes(self) -> None:
         volumes = self.client.list_volumes()
+
+        def extract_volume_status(volume):
+            in_use = len(volume.get("droplet_ids", []) or []) > 0
+            if in_use:
+                return "in-use"
+            else:
+                return "available"
```
Comment (aquamatthias) [inline plugins/digitalocean/resoto_plugin_digitalocean/collector.py:403]:
```suggestion
            return "in-use" if in_use else "available"
```

File: plugins/digitalocean/test/test_collector.py [hunk 2]
```diff
@@ -0,0 +1,345 @@
...
+    cloud_graph.merge(plugin_instance.graph)
+    graph.merge(cloud_graph)
+    sanitize(graph)
+    return graph
+
+
+def check_edges(graph: Graph, from_id: str, to_id: str) -> None:
+    for (node_from, node_to, edge) in graph.edges:
+        if node_from.id == from_id and node_to.id == to_id:
+            return
```
Comment (aquamatthias) [inline plugins/digitalocean/test/test_collector.py:49]: consider to use `graph.has_edge()`

File: plugins/digitalocean/test/test_collector.py [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+
+    check_edges(
+        graph, "do:droplet:289110074", "do:volume:631f81d2-9fc1-11ec-800c-0a58ac14d197"
+    )
+    volume = graph.search_first("id", "do:volume:631f81d2-9fc1-11ec-800c-0a58ac14d197")
+    assert volume.id == "do:volume:631f81d2-9fc1-11ec-800c-0a58ac14d197"
+    assert volume.name == "volume-fra1-01"
+    assert volume.description == "Test volume"
+    assert volume.filesystem_type == "ext4"
+    assert volume.filesystem_label == ""
```
Comment (aquamatthias) [inline plugins/digitalocean/test/test_collector.py:157]: because it has no label? or the label is really an empty string?

**Review by aquamatthias — COMMENTED** [review]
Minor improvement comments. Otherwise lgtm. thanks!

File: plugins/digitalocean/test/test_collector.py (hunk 3 above)
Comment (meln1k) [reply plugins/digitalocean/test/test_collector.py:157]: fixed

File: plugins/digitalocean/test/test_collector.py (hunk 2 above)
Comment (aquamatthias) [reply plugins/digitalocean/test/test_collector.py:49]: Ah - yes the collector uses the whole node as node id - not ideal from my pov. But yes - I see why.
Maybe add a comment, so other people understand why.

**Review by aquamatthias — APPROVED** [review]
thanks

**Pull request merged**